base_classes/NXcg_primitive.nxdl.xml

```diff
@@ -0,0 +1,229 @@
+             For implicit indexing identifiers are defined on the interval
+             :math:`[identifier\_offset, identifier\_offset + c - 1]`.
+             
+             Therefore, implicit identifier are completely defined by the value of
+             identifier_offset and cardinality. For example if identifier run from
+             -2 to 3 the value for identifier_offset is -2.
+             
+             For explicit indexing the field identifier has to be used.
+             Fortran-/Matlab- and C-/Python-style indexing have specific implicit
+             identifier conventions where identifier_offset is 1 and 0 respectively.
+        </doc>
+    </field>
+    <field name="identifier" type="NX_INT">
+        <doc>
+             Identifier of each member for explicit indexing.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="c"/>
+        </dimensions>
+    </field>
+    <field name="center" type="NX_NUMBER" units="NX_ANY">
+        <doc>
+             The center of mass position of each primitive.
+        </doc>
+        <dimensions rank="2">
+            <dim index="1" value="c"/>
+            <dim index="2" value="d"/>
+        </dimensions>
+    </field>
+    <field name="is_center_of_mass" type="NX_BOOLEAN">
+        <doc>
+             True if the center is a center of mass.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="c"/>
+        </dimensions>
+    </field>
+    <field name="shape" type="NX_NUMBER" units="NX_LENGTH">
+        <doc>
+             A qualitative description of the shape of each primitive.
+        </doc>
+        <dimensions rank="2">
+            <dim index="1" value="c"/>
+            <dim index="2" value="d"/>
+        </dimensions>
+    </field>
+    <field name="length" type="NX_NUMBER" units="NX_LENGTH">
+        <doc>
+             Qualifier for the length of characteristic features of the primitive.
+             
+             Often the term length is associated with the assumption that one
+             edge is parallel to an axis of the coordinate system.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="c"/>
+        </dimensions>
+    </field>
+    <field name="width" type="NX_NUMBER" units="NX_LENGTH">
+        <doc>
+             Qualifier often used to describe the length of one characteristic edge
+             within the coordinate system.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="c"/>
+        </dimensions>
+    </field>
+    <field name="is_closed" type="NX_BOOLEAN">
+        <doc>
+             True if primitive is closed such that it has properties like area or volume.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="c"/>
+        </dimensions>
+    </field>
+    <field name="volume" type="NX_NUMBER" units="NX_VOLUME">
+        <doc>
+             Volume of each primitive.
```

Comment:
Is volume an N-D concept?

Reply:
Yes, but then units cannot be NX_VOLUME but NX_ANY